conans/client/toolchain/cmake.py

```diff
@@ -0,0 +1,300 @@
+        endif()
+        set(CONAN_TOOLCHAIN_INCLUDED true)
+
+        ########### Utility macros and functions ###########
+        {{ cmake_macros_and_functions }}
+        ########### End of Utility macros and functions ###########
+
+        # Configure
+        {% if generator_platform %}
+        set(CMAKE_GENERATOR_PLATFORM "{{ generator_platform }}" CACHE STRING "" FORCE){% endif %}
+        {% if toolset %}set(CMAKE_GENERATOR_TOOLSET "{{ toolset }}" CACHE STRING "" FORCE){% endif%}
+
+        # build_type (Release, Debug, etc) is only defined for single-config generators
+        {% if build_type %}
+        set(CMAKE_BUILD_TYPE "{{ build_type }}" CACHE STRING "Choose the type of build." FORCE)
+        {% endif %}
+
+        # --  - CMake.flags --> CMakeDefinitionsBuilder::get_definitions
+        {%- for it, value in definitions.items() %}
+        {%- if it.startswith('CONAN_') %}
+        set({{ it }} "{{ value }}")
+        {%- else %}
+        set({{ it }} "{{ value }}" CACHE STRING "Value assigned from the Conan toolchain" FORCE)
+        {%- endif %}
+        {%- endfor %}
+
+        get_property( _CMAKE_IN_TRY_COMPILE GLOBAL PROPERTY IN_TRY_COMPILE )
+        if(_CMAKE_IN_TRY_COMPILE)
+            message(STATUS "Running toolchain IN_TRY_COMPILE")
+            return()
+        endif()
+
+        message("Using Conan toolchain through ${CMAKE_TOOLCHAIN_FILE}.")
+
+        if(CMAKE_VERSION VERSION_LESS "3.15")
+            message(WARNING
+                " CMake version less than 3.15 doesn't support CMAKE_PROJECT_INCLUDE variable\\n"
+                " used by Conan toolchain to work. In order to get the same behavior you will\\n"
+                " need to manually include the generated file after your 'project()' call in the\\n"
+                " main CMakeLists.txt file:\\n"
+                " \\n"
+                "     project(YourProject C CXX)\\n"
+                "     include(\\"\\${CMAKE_BINARY_DIR}/conan_project_include.cmake\\")\\n"
+                " \\n"
+                " This file contains some definitions and extra adjustments that depend on\\n"
+                " the build_type and it cannot be done in the toolchain.")
+        else()
+            # Will be executed after the 'project()' command
+            set(CMAKE_PROJECT_INCLUDE "{{ conan_project_include_cmake }}")
+        endif()
+
+        # We are going to adjust automagically many things as requested by Conan
+        #   these are the things done by 'conan_basic_setup()'
+         # To support the cmake_find_package generators:
+        {% if cmake_module_path %}
+        set(CMAKE_MODULE_PATH {{ cmake_module_path }} ${CMAKE_MODULE_PATH}){% endif%}
+        {% if cmake_prefix_path %}
+        set(CMAKE_PREFIX_PATH {{ cmake_prefix_path }} ${CMAKE_PREFIX_PATH}){% endif%}
+
+        {% if fpic %}
+        message(STATUS "Conan toolchain: Setting CMAKE_POSITION_INDEPENDENT_CODE=ON (options.fPIC)")
+        set(CMAKE_POSITION_INDEPENDENT_CODE ON)
```

Comment:
not setting to off otherwise?

Reply:
The default is OFF, and shouldn't be messed if shared=True, will be confused to have it OFF for that?

Reply:
yes, I think it sounds logical